### PR TITLE
fastmod: add page

### PR DESCRIPTION
**Pull request by MuhammadSawalhy**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):** v0.4.2

### CODE REVIEW

File: pages.ar/common/fastmod.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# fastmod
+
+> أداة للاستبدال الجزئي للنصوص في قاعدة الأكواد لديك.
+> التعبيرات الاعتيادية تعالجها : <https://crates.io/crates/regex>
```
Comment (git-em) [inline pages.ar/common/fastmod.md:4]: Command descriptions should end in a period
```suggestion
> التعبيرات الاعتيادية تعالجها : <https://crates.io/crates/regex>.

File: pages/common/fastmod.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# fastmod
+
+> A fast partial replacement for the codemod tool, replace and replace all in the whole codebase.
+> Regexes are match with the regex crate: <https://crates.io/crates/regex>.
```
Comment (blueskyson) [inline pages/common/fastmod.md:4]:
```suggestion
> Regexes are matched by Rust regex crate.
```

File: pages/common/fastmod.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern:
+
+`fastmod {{regex_pattern}} {{replacement}} -d {{dir}} --iglob {{wildcard_patterns}}`
+
+- Replace fixed string as it is (not regex), .js or .json files only:
```
Comment (blueskyson) [inline pages/common/fastmod.md:11]:
```suggestion
- Replace fixed non-regex string in .js or .json files:
```

File: pages/common/fastmod.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+`fastmod -F {{match_string}} {{replacement_string}} -e json,js`
+
+- Replace-all without confirmation, replace string as it is (not regex):
```
Comment (blueskyson) [inline pages/common/fastmod.md:15]:
```suggestion
- Replace all with non-regex string and without confirmation:
```

File: pages/common/fastmod.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+`fastmod --accept-all -F {{match_string}} {{replacement_string}}`
+
+- Replace-all similar to the above, but print changed files as well:
```
Comment (blueskyson) [inline pages/common/fastmod.md:19]:
```suggestion
- Replace all like previous example, and print changed files:
```

Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:19]: I want to add a multiline description for one of these examples to make it more clear, how to do that?

Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:19]: Or say two sentences or one long rephrased sentence.

Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:19]: > Replace regex pattern in current directory in all non-ignored files, you can ignore files with .ignore or .gitignore

File: pages/common/fastmod.md (hunk 2 above)
Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:4]: Should I remove the link for the crate? I can add a link for its docs: https://docs.rs/regex/latest/regex/ ?

Comment (git-em) [reply pages/common/fastmod.md]: The convention is using only one link per page

Comment (MuhammadSawalhy) [reply pages/common/fastmod.md]: Sorry, I added the link for the regex crate docs, I think it is better for quick access and to help the `fastmod`'s to get more info about the regex he is writing.

We can assume that 1 is nearly = 2 in this case... 😂

File: pages/common/fastmod.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# fastmod
+
+> A fast partial replacement for the codemod tool, replace and replace all in the whole codebase.
+> Regexes are matched by Rust regex crate: <https://docs.rs/regex/latest/regex>.
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern in current directory in all non-ignored files, you can ignore files with .ignore or .gitignore:
```
Comment (git-em) [inline pages/common/fastmod.md:7]:
```suggestion
- Replace regex pattern in current directory in all files, ignoring files on .ignore and .gitignore:
```

File: pages/common/fastmod.md (hunk 5 above)
Comment (blueskyson) [reply pages/common/fastmod.md]: I just feel `and` is more fitting than `but` in this sentence. Separate this into 2 sentence is fine as well, like `Replace all similar to the above. And print changed files:`

**Comment by MuhammadSawalhy** [comment]
I think the PR is ready to be merged. I hope!

File: pages/common/fastmod.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+> Regexes are matched by Rust regex crate.
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern in current directory in all files, ignoring files on .ignore and .gitignore:
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Replace regex pattern and choose your target files:
+
+`fastmod {{regex}} {{replacement}} -d {{src/}} --iglob {{'**/*.{js,json}'}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:13]: We prefer long arguments if available
```suggestion
`fastmod {{regex}} {{replacement}} --dir {{path/to/directory}} --iglob {{'**/*.{js,json}'}}`
```

File: pages/common/fastmod.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+> A fast partial replacement for the codemod tool, replace and replace all in the whole codebase.
+> Regexes are matched by Rust regex crate.
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern in current directory in all files, ignoring files on .ignore and .gitignore:
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Replace regex pattern and choose your target files:
```
Comment (navarroaxel) [inline pages/common/fastmod.md:11]:
```suggestion
- Replace a regex pattern in all file in a specific directory:
```

File: pages/common/fastmod.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+`fastmod {{regex}} {{replacement}} -d {{src/}} --iglob {{'**/*.{js,json}'}}`
+
+- Replace fixed non-regex string in .js or .json files:
+
+`fastmod -F {{match}} {{replacement}} -e {{json,js}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:17]:
```suggestion
`fastmod --fixed-strings {{match}} {{replacement}} --extensions {{json,js}}`
```

File: pages/common/fastmod.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+- Replace all with non-regex string and without confirmation:
+
+`fastmod --accept-all -F {{match}} {{replacement}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:21]:
```suggestion
`fastmod --accept-all --fixed-strings {{match}} {{replacement}}`
```

File: pages/common/fastmod.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+- Replace regex pattern and choose your target files:
+
+`fastmod {{regex}} {{replacement}} -d {{src/}} --iglob {{'**/*.{js,json}'}}`
+
+- Replace fixed non-regex string in .js or .json files:
+
+`fastmod -F {{match}} {{replacement}} -e {{json,js}}`
+
+- Replace all with non-regex string and without confirmation:
```
Comment (navarroaxel) [inline pages/common/fastmod.md:19]:
```suggestion
- Replace all with non-regex string and without prompt for a confirmation:
```

**Review by blueskyson — APPROVED** [review]

File: pages/common/fastmod.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+- Replace regex pattern in current directory in all files, ignoring files on .ignore and .gitignore:
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Replace a regex pattern in all file in a specific directory:
+
+`fastmod {{regex}} {{replacement}} --dir {{path/to/directory}} --iglob {{'**/*.{js,json}'}}`
+
+- Replace fixed non-regex string in .js or .json files:
```
Comment (navarroaxel) [inline pages/common/fastmod.md:15]: Based on the `grep` page
```suggestion
- Replace an exact string in files with .js and .json extensions (disables regular expressions) :
```

File: pages/common/fastmod.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+`fastmod --fixed-strings {{match}} {{replacement}} --extensions {{json,js}}`
+
+- Replace all with non-regex string and without prompt for a confirmation:
```
Comment (navarroaxel) [inline pages/common/fastmod.md:19]:
```suggestion
- Replace for an exact string without prompt for a confirmation (disables regular expressions):
```

File: pages/common/fastmod.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+`fastmod --accept-all --fixed-strings {{match}} {{replacement}}`
+
+- Replace all like previous example, and print changed files:
+
+`fastmod --accept-all --print-changed-files -F {{match}} {{replacement}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:25]:
```suggestion
`fastmod --accept-all --print-changed-files --fixed-strings {{exact_string}} {{replacement}}`
```

File: pages/common/fastmod.md [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+`fastmod {{regex}} {{replacement}} --dir {{path/to/directory}} --iglob {{'**/*.{js,json}'}}`
+
+- Replace fixed non-regex string in .js or .json files:
+
+`fastmod --fixed-strings {{match}} {{replacement}} --extensions {{json,js}}`
+
+- Replace all with non-regex string and without prompt for a confirmation:
+
+`fastmod --accept-all --fixed-strings {{match}} {{replacement}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:21]:
```suggestion
`fastmod --accept-all --fixed-strings {{exact_string}} {{replacement}}`
```

File: pages/common/fastmod.md [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Replace a regex pattern in all file in a specific directory:
+
+`fastmod {{regex}} {{replacement}} --dir {{path/to/directory}} --iglob {{'**/*.{js,json}'}}`
+
+- Replace fixed non-regex string in .js or .json files:
+
+`fastmod --fixed-strings {{match}} {{replacement}} --extensions {{json,js}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:17]:
```suggestion
`fastmod --fixed-strings {{exact_string}} {{replacement}} --extensions {{json,js}}`
```

File: pages/common/fastmod.md [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+
+- Replace all with non-regex string and without prompt for a confirmation:
+
+`fastmod --accept-all --fixed-strings {{match}} {{replacement}}`
+
+- Replace all like previous example, and print changed files:
```
Comment (navarroaxel) [inline pages/common/fastmod.md:23]:
```suggestion
- Replace for an exact string without prompt for a confirmation, printing changed files:
```

File: pages.ar/common/fastmod.md [hunk 18]
```diff
@@ -0,0 +1,25 @@
+# fastmod
+
+> أداة للاستبدال الجزئي للنصوص في قاعدة الأكواد لديك.
+> التعبيرات الاعتيادية يعالجها قفص من بضاعة رست وهو regex.
+> لمزيد من العلومات: <https://github.com/facebookincubator/fastmod>.
+
+- استبدال التعبيرات الاعتيادية في كل ملفات المسار الحالي وأبنائه في الملفات غير المُتجاهلة بـ .ignore أو .gitignore
+
+`fastmod {{regex_pattern}} {{replacement}}`
```
Comment (navarroaxel) [inline pages.ar/common/fastmod.md:9]: Please translate the placeholders for get this easier for non-English speakers. Like `path/to/directory` and `regex_pattern`.

File: pages/common/fastmod.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+> Regexes are matched by Rust regex crate.
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern in all files of the current directory, ignoring files on .ignore and .gitignore:
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Case-insensitively replace regex pattern in specific files even if ignored, or directory ignoring files on .ignore and .gitignore:
+
+`fastmod --ignore-case {{regex_pattern}} {{replacement}} -- {{file}} {{dir/}} {{...}}`
```
Comment (navarroaxel) [inline pages/common/fastmod.md:13]:
```suggestion
`fastmod --ignore-case {{regex_pattern}} {{replacement}} -- {{path/to/file1}} {{path/to/file2}} {{path/to/directory}}`
```

File: pages.ar/common/fastmod.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+> أداة للاستبدال الجزئي للنصوص في قاعدة الأكواد لديك.
+> التعبيرات النمطية يعالجها قفص من بضاعة رست وهو regex.
+> لمزيد من العلومات: <https://github.com/facebookincubator/fastmod>.
+
+- استبدال بالتعبيرات النمطية في كل ملفات المسار الحالي وأبنائه في الملفات غير المُتجاهلة بـ .ignore أو .gitignore
+
+`fastmod {{تعبير_نمطي}} {{بديل}}`
+
+- استبدال متجاهلا حالة الحرف في ملف حتى لو مُتجاهَلٌ أو في ملفات مسار مستبعدا المُتجاهَل منها بـ .ignore أو .gitignore
+`fastmod --ignore-case {{regex_pattern}} {{replacement}} -- {{file}} {{dir/}} {{...}}`
```
Comment (navarroaxel) [inline pages.ar/common/fastmod.md:12]: placeholders here

File: pages/common/fastmod.md [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+
+> A fast partial replacement for the codemod tool, replace and replace all in the whole codebase.
+> Regexes are matched by Rust regex crate.
+> More information: <https://github.com/facebookincubator/fastmod>.
+
+- Replace regex pattern in all files of the current directory, ignoring files on .ignore and .gitignore:
+
+`fastmod {{regex_pattern}} {{replacement}}`
+
+- Case-insensitively replace regex pattern in specific files even if ignored, or directory ignoring files on .ignore and .gitignore:
```
[the remark this answers is not in the document]
Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:11]: The new description doesn't clarify that if you specify a file it will be included even if it is ignore in `.ignore` or `.gitignore`

File: pages.ar/common/fastmod.md (hunk 20 above)
Comment (MuhammadSawalhy) [reply pages.ar/common/fastmod.md:12]: > placeholders here

In place of the ellipsis `...`?

File: pages/common/fastmod.md (hunk 19 above)
Comment (MuhammadSawalhy) [reply pages/common/fastmod.md:13]: `...` is more concise and meaningful. Can I keep it?

Comment (navarroaxel) [reply pages/common/fastmod.md:13]: :thinking: we use this for enumerations:
```suggestion
`fastmod --ignore-case {{regex_pattern}} {{replacement}} -- {{path/to/file_or_directory1 path/to/file_or_directory2 ...}}`
```

File: pages/common/fastmod.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (navarroaxel) [reply pages/common/fastmod.md:11]: It says `ignoring files on` :thinking:  maybe this is better:
```suggestion
- Replace a regex pattern in case-insensitive mode in specific files or directories, ignoring .ignore and .gitignore:
```

[the remark this answers is not in the document]
Comment (MuhammadSawalhy) [reply pages/common/fastmod.md]: > Replace a regex pattern in a case-insensitive mode in specific files or directories:

I think this is a better description because all these examples files are ignored using .ignore or .gitignore which is emphasized in the first example.

**Comment by MuhammadSawalhy** [comment]
OK, I got it wrong again

> should contain **_no_** trailing whitespace

I've used this command to check before commiting:

```
tldr-lint pages/common/fastmod.md
```

**Review by CleanMachine1 — APPROVED** [review]

**Review by navarroaxel — APPROVED** [review]
